Classes/Service/CacheApiService.php

```diff
@@ -203,4 +192,50 @@ public function clearAllExceptPageCache() {
 	protected function clearAllActiveOpcodeCacheWrapper($fileAbsPath) {
 		\TYPO3\CMS\Core\Utility\OpcodeCacheUtility::clearAllActive($fileAbsPath);
 	}
+
+	/**
+	 * This clear cache implementation follows a pretty brutal approach.
+	 * Goal is to reliably get rid of cache entries, even if some broken
+	 * extension is loaded that would kill the backend 'clear cache' action.
+	 *
+	 * Therefor this method "knows" implementation details of the cache
+	 * framework and uses them to clear all file based cache (typo3temp/Cache)
+	 * and database caches (tables prefixed with cf_) manually.
+	 *
+	 * This is the change to the InstallTool->clearAll() method
+	 * This will not happened (so we do not respect different Cache Backends)
+	 *
+	 * After that ext_tables and ext_localconf of extensions are loaded, those
+	 * may register additional caches in the caching framework with different
+	 * backend, and will then clear them with the usual flush() method.
+	 *
+	 * @return void
+	 * @see \TYPO3\CMS\Install\Service\ClearCacheService
+	 */
```

Comment:
see the original-Method. After flushDirectory('typo3temp/Cache') the File-Based-Caching-Framework ist completly inconsistence, because all Implementations are Singletons which assure on Constructions that their Caching-Folder exists